nucleus/dataset.py

```diff
@@ -155,12 +156,61 @@ def model_runs(self) -> List[str]:
         return response
 
     @property
-    def slices(self) -> List[str]:
+    def slices(self) -> List[Slice]:
         """List of all Slice IDs created from the Dataset."""
         response = self._client.make_request(
             {}, f"dataset/{self.id}/slices", requests.get
         )
-        return response
+        return [Slice.from_request(info, self._client) for info in response]
+
+    def get_slice_by_name(self, name: str) -> Slice:
+        """Returns a slice object with the given name.
+
+        Parameters:
+            name: Name of the desired slice to look up.
+
+        Raises:
+            NotFound if the name does not exist
+
+        Returns:
+            :class:`Slice`: The Nucleus slice as an object.
+        """
+        response = self._client.make_request(
+            {}, f"dataset/{self.id}/slices?name={name}", requests.get
+        )
+        if len(response) == 0:
+            raise NotFoundError(f"Slice with name {name} was not found.")
+
+        # slice names are unique per dataset, so we can guarantee len is 1 here
+        return Slice.from_request(response[0], self._client)
+
+    def get_slices_by_type(
```

Comment:
This is a little bit to `scaleapi-ey` let's just have a single `get_slice` with `kwargs` for query parameters.